app/src/main/java/com/google/samples/apps/sunflower/compose/PlantDetailView.kt

```diff
@@ -0,0 +1,505 @@
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        val iconModifier = Modifier.sizeIn(maxWidth = 32.dp, maxHeight = 32.dp)
+            .drawBackground(color = Color.White, shape = CircleShape)
+
+        IconButton(
+            onClick = onBackClicked,
+            modifier = Modifier.padding(start = 12.dp).plus(iconModifier)
+        ) {
+            Icon(Icons.Filled.ArrowBack)
+        }
+        val shareAccessibilityLabel = stringResource(R.string.menu_item_share_plant)
+        IconButton(
+            onClick = onShareClicked,
+            modifier = Modifier.padding(end = 12.dp).plus(iconModifier).semantics {
+                accessibilityLabel = shareAccessibilityLabel
+            }
+        ) {
+            Icon(Icons.Filled.Share)
+        }
+    }
+}
+
+@Composable
+private fun PlantInformation(
+    name: String,
+    wateringInterval: Int,
+    description: String,
+    onNamePositioned: (LayoutCoordinates) -> Unit,
+    toolbarShown: Boolean
+) {
+    Box(modifier = Modifier.padding(24.dp)) {
+        Hide(toolbarShown) { hideModifier ->
+            Text(
+                text = name,
+                style = MaterialTheme.typography.h5,
+                modifier = Modifier.padding(start = 8.dp, end = 8.dp, bottom = 16.dp)
+                    .gravity(Alignment.CenterHorizontally).onPositioned {
+                        onNamePositioned(it)
+                    }.plus(hideModifier)
+            )
+        }
+        Text(
+            text = stringResource(id = R.string.watering_needs_prefix),
+            color = MaterialTheme.colors.primaryVariant,
+            style = MaterialTheme.typography.body1.copy(
+                fontWeight = FontWeight.Bold
+            ),
+            modifier = Modifier.padding(horizontal = 8.dp).gravity(Alignment.CenterHorizontally)
+        )
+        Text(
+            text = getQuantityString(R.plurals.watering_needs_suffix, wateringInterval),
+            color = MaterialTheme.colors.onBackground.copy(alpha = 0.6f),
+            modifier = Modifier.padding(start = 8.dp, end = 8.dp, bottom = 16.dp)
+                .gravity(Alignment.CenterHorizontally)
+        )
+        PlantDescription(description)
+    }
+}
+
+@Composable
+private fun PlantDescription(description: String) {
+    // Issue: User input doesn't work properly, HTML links cannot be clicked - b/158088138
+    // HTML support coming to Compose - b/139320905
+    AndroidView(resId = R.layout.item_plant_description) {
+        (it as TextView).text = HtmlCompat.fromHtml(description, HtmlCompat.FROM_HTML_MODE_COMPACT)
+    }
+}
+
+/**
+ * Gives a parallax effect to the content at the top of a [VerticalScroller].
+ */
+@Composable
+private fun ParallaxEffect(
+    scrollerPosition: ScrollerPosition,
+    modifier: Modifier = Modifier,
+    content: @Composable (Modifier) -> Unit
+) {
+    val offset = scrollerPosition.value / ParallaxDelta
```

Comment:
This offset calc is duplicated in `getFabOffset` can we hoist it instead to avoid this/potential drifting?

Reply:
They're used in different places, that's why I wanted to keep them in a constant. Hoisting this would be confusing and too much IMO

Reply:
Sorry, misunderstood your comment! Let's see what I can do :) thanks!

Reply:
Created a new extension function on density that gives you that offset in Dps! Thanks